dev/release/verify-release-candidate.sh

```diff
@@ -616,14 +616,12 @@ test_and_install_cpp() {
     -DPARQUET_REQUIRE_ENCRYPTION=ON \
     ${ARROW_CMAKE_OPTIONS:-} \
     ${ARROW_SOURCE_DIR}/cpp
-  cmake --build . --target install
+  cmake --build . --target install -j$NPROC
```

Comment:
`-j` doesn't exist in CMake 3.11 or earlier:  https://cmake.org/cmake/help/v3.11/manual/cmake.1.html

How about defining `CMAKE_BUILD_PARALLEL_LEVEL` environment variable instead?

https://cmake.org/cmake/help/latest/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html#envvar:CMAKE_BUILD_PARALLEL_LEVEL

```shell
  export CMAKE_BUILD_PARALLEL_LEVEL=${CMAKE_BUILD_PARALLEL_LEVEL:-$(nproc)}
  cmake --build ...
```